catchup/catchpointService.go

```diff
@@ -803,3 +812,26 @@ func (cs *CatchpointCatchupService) initDownloadPeerSelector() {
 			})
 	}
 }
+
+// checkLedgerDownload sends a HEAD request to the ledger endpoint of peers to validate the catchpoint's availability
+// before actually starting the catchup process.
+// The error returned is either from an unsuccessful request or a successful request that did not return a 200.
+func (cs *CatchpointCatchupService) checkLedgerDownload() error {
+	round, _, err := ledgercore.ParseCatchpointLabel(cs.stats.CatchpointLabel)
+	if err != nil {
+		return cs.abort(fmt.Errorf("failed to parse catchpoint label : %v", err))
+	}
+	peerSelector := makePeerSelector(cs.net, []peerClass{{initialRank: peerRankInitialFirstPriority, peerClass: network.PeersPhonebookRelays}})
```

Comment:
I don't know a lot about the catchup service but I see that we have other ways of initializing the `peerSelector.peerClasses` apart from only **Relays** and that relays aren't usually given **First Priority**. eg: https://github.com/algorand/go-algorand/blob/91771c09c2d425097d23f1179986fcb91d8c4302/catchup/service.go#L812

I'm curious of the reasoning here.